AJ-1749 Remove import-service from firecloud-orchestration

jladieu opened the pull request:
Remove import service.  Once https://github.com/broadinstitute/terra-helmfile/pull/5556 is live, cWDS will effectively field 100% of import traffic and all import-service backend resources will be deactivated.

Process:

* Rename ImportServiceTypes -> LegacyServiceTypes (IDE refactor only)
* Move `LegacyFileTypes` to `CwdsDao` (Manual refactor)
* Rename ImportServiceDAO methods to match CwdsDao (IDE refactor only)
* Make `MockCwdsDao` like `MockImportServiceDao` (manual)
* Eliminate `importServiceDao.getJobV1` (manual)
* Eliminate `importServiceDao.listJobsV1` (manual)
* Remove `importV1` from ImportService: and then make all the tests
still pass for vaguely the same reasons as before, except using cWDS.
* Prune references to ImportService
* Make supportedFormats consistent

Related PRs:
* terra-helmfile: https://github.com/broadinstitute/terra-helmfile/pull/5538
* rawls: https://github.com/broadinstitute/rawls/pull/2866

I, the developer opening this PR, do solemnly pinky swear that:

- [x] ~I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes~ (N/A)
- [x] ~I've updated the RC_XXX release ticket with any manual steps required to release this change~ (N/A)
- [x] ~I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing~ (N/A)

In all cases:

- [x] Get two thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green
- [x] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [x] Test this change deployed correctly and works on dev environment after deployment

jladieu commented on src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockRawlsDAO.scala:
```diff
@@ -271,7 +273,7 @@ class MockRawlsDAO extends RawlsDAO {
       namespace = "namespace",
       name = "name",
-      workspaceId = "workspaceId",
+      workspaceId = mockWorkspaceId,
```
Needed to switch this to a UUID to prevent an error attempting to initialize an actual `UUID` from the `workspaceId`

jladieu commented on src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockRawlsDAO.scala:
```diff
@@ -15,12 +15,14 @@ import org.broadinstitute.dsde.rawls.model._
 // Common things that can be accessed from tests
 object MockRawlsDAO {
 
+  val mockWorkspaceId = UUID.randomUUID().toString
```
Made available for a couple tests to refer to it.

jladieu commented on src/test/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceApiServiceSpec.scala:
```diff
@@ -1109,21 +1103,20 @@ class WorkspaceApiServiceSpec extends BaseServiceSpec with WorkspaceApiService w
           ~> sealRoute(workspaceRoutes)) ~> check {
-            status should equal(BadRequest)
-            responseAs[String] should include ("Bad request as reported by import service")
+            status should equal(InternalServerError) // TODO: bubble up 400 / BadRequest
```
The tests verify _current_ behavior in prod, however these tests used to verify what happened in the interaction to `ImportService` so I left these TODOs in place to trigger a conversation over whether we should continue to strive to resurface errors, or if we're okay with just bubbling up as `InternalServerError`.  If we're okay with leaving this as is, I could get rid of the TODOs.

nawatts replied:
My vote would be to deprecate Orchestration's import related endpoints entirely and move everything to talk directly to CWDS.

davidangb replied:
For these APIs, I would much rather bubble up the actual error/response code instead of wrapping everything in a 500. Since this current PR does not change behavior, I'm fine making that change in a follow-on.

jladieu replied:
>For these APIs, I would much rather bubble up the actual error/response code instead of wrapping everything in a 500. Since this current PR does not change behavior, I'm fine making that change in a follow-on.

I plan to leave out any changes related to error handling from this PR, but wanted to highlight @nawatts 's suggestion here which is worth considering: https://github.com/broadinstitute/firecloud-orchestration/pull/1355#discussion_r1598481587

jladieu commented on src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockCwdsDAO.scala:
```diff
@@ -1,34 +1,74 @@
+      workspaceId: String,
+      importRequest: AsyncImportRequest
+  )(implicit userInfo: UserInfo): GenericJob = {
+    importRequest.filetype match {
```
Copied this logic over from `MockImportServiceDAO` and then swapped this in wherever tests relied on it when interacting with `ImportService`.

jladieu commented on src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/CwdsDAO.scala:
```diff
@@ -1,9 +1,15 @@
 import org.databiosphere.workspacedata.client.ApiException
 import org.databiosphere.workspacedata.model.GenericJob
 
+object LegacyFileTypes {
```
Pulled over from `ImportServiceDAO` which went away.

nawatts replied:
👍 for calling these out as for backwards compatibility

jladieu committed:
Remove import service:

Process:

* Rename ImportServiceTypes -> LegacyServiceTypes (IDE refactor only)
* Move `LegacyFileTypes` to `CwdsDao` (Manual refactor)
* Rename ImportServiceDAO methods to match CwdsDao (IDE refactor only)
* Make `MockCwdsDao` like `MockImportServiceDao` (manual)
* Eliminate `importServiceDao.getJobV1` (manual)
* Eliminate `importServiceDao.listJobsV1` (manual)
* Remove `importV1` from ImportService: and then make all the tests
still pass for vaguely the same reasons as before, except using cWDS.
* Prune references to ImportService
* Make supportedFormats consistent

calypsomatic commented on src/test/scala/org/broadinstitute/dsde/firecloud/EntityServiceSpec.scala:
```diff
@@ -111,62 +124,69 @@ class EntityServiceSpec extends BaseServiceSpec with BeforeAndAfterEach {
+      }
     }
 
     "should return error for (async=true) when import service returns sync error" in {
```
Test description should be "cwds" now, right?

jladieu replied:
Thanks for noticing this!  This triggered an additional round of find/replace and I _think_ I got them all this time.

calypsomatic commented on src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockCwdsDAO.scala:
```diff
@@ -1,34 +1,73 @@
+      else if (importRequest.url.contains("good")) makeJob(workspaceId)
+      else
+        throw new ApiException(
+          EnhanceYourCalm.intValue,
```
TIL

jladieu committed:
Rename various `{ImportService->Cwds}` references.

* IDE: `{ImportService->Cwds}ListResponse`
* IDE: `{ImportService->Cwds}Response`
* Manual/IDE: Various methods/comments

calypsomatic commented on automation/README.md:
```diff
@@ -38,7 +38,7 @@ TODO: update `render-local-env.sh` so it doesn't require manual code changes
 
 ##### Testing against your own FiaB
 
```
Not fully relevant to this PR, but FiaBs don't exist anymore

davidangb commented on local-dev/templates/firecloud-orchestration.conf.ctmpl:
```diff
@@ -143,10 +143,3 @@ staticNotebooks {
-  server = "https://terra-importservice-dev.appspot.com"
-  bucketName = "import-service-batchupsert-dev"
-  enabled = true
-}
```
do you have a corresponding terra-helmfile PR to remove this stanza?

jladieu replied:
[github.com/broadinstitute/terra-helmfile/pull/5538](https://github.com/broadinstitute/terra-helmfile/pull/5538/files)

jladieu commented:
>For these APIs, I would much rather bubble up the actual error/response code instead of wrapping everything in a 500. Since this current PR does not change behavior, I'm fine making that change in a follow-on.

I plan to leave out any changes related to error handling from this PR, but wanted to highlight @nawatts 's suggestion here which is worth considering: https://github.com/broadinstitute/firecloud-orchestration/pull/1355#discussion_r1598481587
